sqlbrite/src/androidTest/java/com/squareup/sqlbrite/SqlBriteContentProviderTests.java

```diff
@@ -0,0 +1,191 @@
+    db.setLoggingEnabled(false);
+    db.setLogger(new SqlBrite.Logger() {
+      @Override public void log(String message) {
+        logs.add(message);
+      }
+    });
+
+    getMockContentResolver().insert(TABLE, values("key1", "value1"));
+    assertThat(logs).isEmpty();
+  }
+
+  public void testCreateQueryObservesInsert() {
+    subscription = db.createQuery(TABLE, null, null, null, null, false).subscribe(o);
+    o.assertCursor().isExhausted();
+
+    getMockContentResolver().insert(TABLE, values("key1", "val1"));
+    o.assertCursor().hasRow("key1", "val1").isExhausted();
+  }
+
+  public void testCreateQueryObservesUpdate() {
+    getMockContentResolver().insert(TABLE, values("key1", "val1"));
+    subscription = db.createQuery(TABLE, null, null, null, null, false).subscribe(o);
+    o.assertCursor().hasRow("key1", "val1").isExhausted();
+
+    getMockContentResolver().update(TABLE, values("key1", "val2"), null, null);
+    o.assertCursor().hasRow("key1", "val2").isExhausted();
+  }
+
+  public void testCreateQueryObservesDelete() {
+    getMockContentResolver().insert(TABLE, values("key1", "val1"));
+    subscription = db.createQuery(TABLE, null, null, null, null, false).subscribe(o);
+    o.assertCursor().hasRow("key1", "val1").isExhausted();
+
+    getMockContentResolver().delete(TABLE, null, null);
+    o.assertCursor().isExhausted();
+  }
+
+  public void testUnsubscribeDoesNotTrigger() {
+    final List<String> logs = new ArrayList<>();
+    db.setLogger(new SqlBrite.Logger() {
+      @Override public void log(String message) {
+        logs.add(message);
+      }
+    });
+
+    subscription = db.createQuery(TABLE, null, null, null, null, false).subscribe(o);
+    o.assertCursor().isExhausted();
+    subscription.unsubscribe();
+
+    getMockContentResolver().insert(TABLE, values("key1", "val1"));
+    o.assertNoMoreEvents();
+    assertThat(logs).isEmpty();
+  }
+
+  private ContentValues values(String key, String value) {
+    ContentValues result = new ContentValues();
+    result.put(KEY, key);
+    result.put(VALUE, value);
+    return result;
+  }
+
+  public static final class TestContentProvider extends MockContentProvider {
+    private final Map<String, String> storage = new LinkedHashMap<>();
+
+    private ContentResolver contentResolver;
+
+    void init(ContentResolver contentResolver) {
+      this.contentResolver = contentResolver;
+    }
+
+    @Override public Uri insert(Uri uri, ContentValues values) {
+      storage.put(values.getAsString(KEY), values.getAsString(VALUE));
+      contentResolver.notifyChange(uri, null);
+      return Uri.parse(AUTHORITY + "/" + values.getAsString(KEY));
+    }
+
+    @Override public int update(Uri uri, ContentValues values, String selection,
+        String[] selectionArgs) {
+      for (String key : storage.keySet()) {
+        storage.put(key, values.getAsString(VALUE));
+      }
+      contentResolver.notifyChange(uri, null);
+      return storage.size();
```

Comment:
I think that this technically violates the method spec, but I don't think it matters.